worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,68 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e) {
+    if (!e.application.getInputMessageProcessor().getProcessSubdocumentsSeparately() && e.rootDocument.hasSubdocuments()) {
```

Comment:
The ` && e.rootDocument.hasSubdocuments()` is not required.

Reply:
If I do not use it, I think I will get an exception if the document does not have subdocuments.

Reply:
Why?

Reply:
removed